src/main/java/cd/go/authorization/github/executors/SearchUsersRequestExecutor.java

```diff
@@ -0,0 +1,82 @@
+package cd.go.authorization.github.executors;
+
+
+import cd.go.authorization.github.GitHubClientBuilder;
+import cd.go.authorization.github.models.AuthConfig;
+import cd.go.authorization.github.models.User;
+import cd.go.authorization.github.utils.Util;
+import com.thoughtworks.go.plugin.api.request.GoPluginApiRequest;
+import com.thoughtworks.go.plugin.api.response.DefaultGoPluginApiResponse;
+import com.thoughtworks.go.plugin.api.response.GoPluginApiResponse;
+import org.kohsuke.github.GHUser;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.PagedSearchIterable;
+
+import java.io.IOException;
+import java.util.*;
+
+import static cd.go.authorization.github.GitHubPlugin.LOG;
+
+public class SearchUsersRequestExecutor implements RequestExecutor {
+
+    private static final String SEARCH_TERM = "search_term";
+    private GoPluginApiRequest request;
+    private GitHubClientBuilder gitHubClientBuilder;
+
+    public SearchUsersRequestExecutor(GoPluginApiRequest request) {
+        this(request, new GitHubClientBuilder());
+    }
+
+    SearchUsersRequestExecutor(GoPluginApiRequest request, GitHubClientBuilder gitHubClientBuilder) {
+        this.request = request;
+        this.gitHubClientBuilder = gitHubClientBuilder;
+    }
+
+    SearchUsersRequestExecutor(GitHubClientBuilder gitHubClientBuilder) {
+        this.gitHubClientBuilder = gitHubClientBuilder;
+    }
+
+    @Override
+    public GoPluginApiResponse execute() throws Exception {
+        String requestJson = request.requestBody();
+        Map<String, String> requestParam = Util.GSON.fromJson(requestJson, Map.class);
+        String searchTerm = requestParam.get(SEARCH_TERM);
+        List<AuthConfig> authConfigs = AuthConfig.fromJSONList(requestJson);
+
+        final Set<User> users = searchUsers(searchTerm, authConfigs);
+
+        return new DefaultGoPluginApiResponse(200, Util.GSON.toJson(users));
+    }
+
+    private Set<User> searchUsers(String searchTerm, List<AuthConfig> authConfigs) throws IOException {
+        final HashSet<User> users = new HashSet<>();
+        if (authConfigs != null || !authConfigs.isEmpty()) {
+            for (AuthConfig authConfig : authConfigs) {
+                try {
```

Comment:
Are you planning to limit search result for a AuthConfig or for all authconfig ?

Reply:
Not sure about that. Right now, its limited per auth config. Not sure which is better or if it will improve performance.

Reply:
Mostly per GoCD server user will have only one authconfig. I think it will be fine limiting 10 result per authconfig